test/metric.jl

```diff
@@ -25,8 +26,12 @@ using ReTest, Random, AdvancedHMC
         rng = MersenneTwister(1)
         θ = randn(rng, D)
         ℓπ(θ) = 1
-        for metric in
-            [UnitEuclideanMetric(1), DiagEuclideanMetric(1), DenseEuclideanMetric(1)]
+        for metric in [
+            UnitEuclideanMetric(1),
+            DiagEuclideanMetric(1),
+            DenseEuclideanMetric(1),
+            RankUpdateEuclideanMetric(1),
```

Comment:
This seems to test only the constructor and resizing. Could we have a more extensive test that actually uses `RankUpdateEuclideanMetric` in a sampler and hits all the key methods?